### DISCUSSION TITLE
Gamelift in AWS Java SDK is not working when packed with webpack and use in browser

### DISCUSSION THREAD

**Discussion by ruellm** [aws/aws-sdk-js-v3#3567] [General]
## Describe the bug
Gamelift module of the V3 SDK seems does not work when packaged with webpack.

#### SDK version number
   "@aws-sdk/client-gamelift": "^3.53.0",
    GameLiftLocal-1.0.5 from GameLift_06_03_2021 release

#### Is the issue in the browser/Node.js/ReactNative?
Browser

## Steps to reproduce
I am using Gamelift Local server running in port 9080.
GameLiftLocal-1.0.5 from GameLift_06_03_2021

posted stackoverflow report: 
https://stackoverflow.com/questions/71346019/using-aws-javascript-sdk-v3-for-gamelift-browser

I uploaded the source code
https://drive.google.com/file/d/17mHF83gp5cBAYQEgsgKfRk712pOAkfNe/view?usp=sharing

folders:
V2Testing - contains an expressJS and the website that loads the bundle.js, It was named V2 as I was testing V2 min.js before.
V3 - is the standalone nodejs application to test it
WebpackTestingV3 - contains the nodejs + webpacked items

#### Observed behavior
error 'Credentials missing' in browser while in straight up nodejs test it is succesful.
I changed implementation to use the AWS.Credentials and now im getting the "TypeError: Failed to fetch".

#### Expected behavior
It should behave similar to nodejs.

## Additional context
see stackoverflow report 
https://stackoverflow.com/questions/71346019/using-aws-javascript-sdk-v3-for-gamelift-browser

**Comment by AllanZhengYP** [comment]
Hi @ruellm

I think this should be a v2 SDk question not v3 one. In the code snippet, I find you set the credential incorrectly: the credentials should be supplied to the `credentials` parameter of the constroctor options hash:
```javascript
const gamelift = new AWS.GameLiftClient({
        endpoint : 'http://localhost:9080',
        credentials: {
                accessKeyId :'abcde',
                secretAccessKey:'abcde',
        },
        region:'us-west-2'
});
```
Here the [v2 get-started doc link](https://docs.aws.amazon.com/sdk-for-javascript/v2/developer-guide/global-config-object.html)
